src/test/java/de/medizininformatikinitiative/torch/rest/ExtractDataParametersParserTest.java

```diff
@@ -0,0 +1,90 @@
+package de.medizininformatikinitiative.torch.rest;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.parser.DataFormatException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.medizininformatikinitiative.torch.model.crtdl.Crtdl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+
+class ExtractDataParametersParserTest {
+
+    private ExtractDataParametersParser extractDataParametersParser;
+    private FhirContext fhirContext;
+
+    @BeforeEach
+    void setup() {
+        fhirContext = FhirContext.forR4();
+        extractDataParametersParser = new ExtractDataParametersParser(fhirContext, new com.fasterxml.jackson.databind.ObjectMapper());
+    }
+
+    @Test
+    void parseCrtdl_emptyParameters_throws() {
+        String emptyParamsJson = "{\"resourceType\":\"Parameters\",\"parameter\":[]}";
+
+        assertThatThrownBy(() -> extractDataParametersParser.parseCrtdl(emptyParamsJson))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("Empty Parameters");
+    }
+
+    @Test
+    void parseCrtdl_missingCrtdlParameter_returnsErrorMono() {
+        String noCrtdlJson = """
+                {
+                  "resourceType": "Parameters",
+                  "parameter": [
+                    {
+                      "name": "patient",
+                      "valueString": "patient-123"
+                    }
+                  ]
+                }
+                """;
+
+        assertThatThrownBy(() -> extractDataParametersParser.parseCrtdl(noCrtdlJson))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("No base64 encoded CRDTL content found in Parameters resource");
+    }
+
+    @Test
+    void parseCrtdl_invalidJson_returnsErrorMono() {
+        String invalidJson = "this is not valid JSON";
+
+        assertThatThrownBy(() -> extractDataParametersParser.parseCrtdl(invalidJson))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasCauseInstanceOf(DataFormatException.class);
+    }
+
+    @Test
+    void parseCrtdl_ioException_returnsErrorMono() throws IOException {
+        ObjectMapper mockObjectMapper = Mockito.mock(ObjectMapper.class);
```

Comment:
the IOException is mostly a JsonParsingException. Please just use invalid JSON here